qiskit_experiments/framework/experiment_data.py

```diff
@@ -44,12 +56,18 @@ def __init__(self, experiment=None, backend=None, parent_id=None, job_ids=None):
             metadata=experiment._metadata() if experiment else {},
         )
 
+        # Add component data and set parent ID to current container
+        self._child_data = ThreadSafeList()
```

Comment:
I think these should be used anytime a container is appended or modified during an async job in a way that means it could potentially be modified by different threads running in parallel.